web/src/settings_panel_menu.js

```diff
@@ -64,22 +66,39 @@ export class SettingsPanelMenu {
 
     show() {
         this.$main_elem.show();
-        const section = this.current_tab();
+        const section = this.current_tab;
+        const user_settings_tab = this.current_user_settings_tab;
         if (two_column_mode()) {
             // In one column mode want to show the settings list, not the first settings section.
-            this.activate_section_or_default(section);
+            this.activate_section_or_default(section, user_settings_tab);
         }
         this.$curr_li.trigger("focus");
+        this.org_user_settings_toggler = components.toggle({
+            html_class: "fuck",
+            child_wants_focus: true,
+            values: [
+                {label: $t({defaultMessage: "Users"}), key: "active-users-admin"},
+                {
+                    label: $t({defaultMessage: "Deactivated users"}),
+                    key: "deactivated-users-admin",
+                },
+                {label: $t({defaultMessage: "Invitations"}), key: "invites-list-admin"},
+            ],
+            callback: (_name, key) => {
+                browser_history.update(`#organization/user-list-admin/${key}`);
+                this.set_user_settings_tab(key);
+                $(".user-settings-section").hide();
+                $(`[data-user-settings-section="${CSS.escape(key)}"]`).show();
+            },
+        });
```

Comment:
Initially, I had this defined in the constructor. But this lead to the toggle cliick listeners not being initialized when trying to access the settings through the gear menu from the zulip home page.

Reply:
I guess it would be better to just have this code in `settings_users.set_up_humans` as this is specifically related to the users list. And by keeping this here, this code is executed everytime the user toggles between `Personal` and `Organization` tabs in the left, which is not necessary.

Reply:
That was how it was in the original PR, yes. But `settings_panel_menu` imports `settings_sections` which imports `settings_users`. But the toggle's callback requires `settings_panel_menu` in `settings_users` causing a circular import.

`settings_panel_menu` felt like the next best place to put it while avoiding the circular import issue.

Reply:
Figured it out, introduced 98539623aa7b8828723ddc99eca188505cd463a6 to expose `register_event_handlers` method for the toggle component. I've moved the toggle declaration to the constructor constructor and the events are reregistered only in the case when the `section` is `user-list-admin`.

This should be ready for another review.

Reply:
I am seeing a bug where the tabs are not visible when I open the settings UI from the gear menu first time after a reload.

Reply:
Thanks, good catch. Fixed [here](https://github.com/zulip/zulip/compare/a0d6b06a914e17d24f6890d2f23f90190d59bbf6..345a693ab9d4440c1b98d40b044ae72858cf2e30) and [here](https://github.com/zulip/zulip/compare/345a693ab9d4440c1b98d40b044ae72858cf2e30..4e09d5a908b4d7c7c75a93bfcb137d95f6528ff1)